src/sha2.rs

```diff
@@ -1,37 +1,8 @@
 #[cfg(not(openssl))]
-use sha2::{Digest, Sha256 as Ctx};
+pub use sha2::{Digest, Sha256};
```

Comment:
this is another problem. In order to call `finalize()` on the `sha2::Sha256` object, the `Hasher` trait needs to be in scope. i.e. users need to import it. However, this won't be available when using the openssl backend.